spring-grpc-core/src/main/java/org/springframework/grpc/client/DefaultGrpcChannelFactory.java

```diff
@@ -72,6 +72,10 @@ public void setCredentialsProvider(ChannelCredentialsProvider credentials) {
 
 	@Override
 	public ManagedChannelBuilder<?> createChannel(String authority) {
+		String target = this.targets.getTarget(authority);
```

Comment:
So `this.targets.getTarget` ultimately goes into NamedVirtualTargets in the auto-config:
```java
		@Override
		public String getTarget(String authority) {
			NamedChannel channel = this.channels.getChannel(authority);
			return this.channels.getTarget(channel.getAddress());
		}
```
which ultimately still calls `this.channels.getChannel` and `this.channels.getTarget` which in turn still does the duplicate mapping. I don't think this code is doing what you think it is.

So far this PR seems to only be moving the NamedChannel out of the config props and into core.

Reply:
I would prefer the first thing we do is surface the duplicate entries issue in a test and get that figured out and fixed and then look into moving the NamedChannel into core.

Reply:
BTW - here is a rough prototype of how I would split the named channel out of autconfgure module. https://github.com/spring-projects-experimental/spring-grpc/pull/74

Note that something similar could be done for the NamedCredentials. 

@dsyer wdyt?